tests/test_appdynamics.py

```diff
@@ -239,13 +319,25 @@ def test_appdynamics_healthrule_violations_errors(monkeypatch, fx_exception):
 
     monkeypatch.setattr('requests.Session.get', get)
 
-    url = 'https://appdynamics'
-    application = 'App 1'
+    cli = AppdynamicsWrapper(URL, username=USER, password=PASS)
+
+    with pytest.raises(raised):
+        cli.healthrule_violations(APPLICATION, time_range_type=BEFORE_NOW, duration_in_mins=5)
+
+
+def test_appdynamics_metric_data_errors(monkeypatch, fx_exception):
```

Comment:
I suggest adding another test case for invalid kwargs. That way we can assert kwargs are validated in ``metric_data`` and easily detect regressions. Similar to ``test_appdynamics_healthrule_violations_kwargs_error``